appengine/genetics/js/cage.js

```diff
@@ -0,0 +1,730 @@
+/**
+ * Blockly Games: Genetics
+ *
+ * Copyright 2016 Google Inc.
+ * https://github.com/google/blockly-games
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview JavaScript for the game logic for the Genetics game.
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+'use strict';
+
+goog.provide('Genetics.Cage');
+
+goog.require('Genetics.Mouse');
+
+/**
+ * The maximum population in a cage.
+ * @type {number}
+ */
+Genetics.Cage.MAX_POPULATION = 10;
+
+/**
+ * The number of mice added to the cage for each player at the start of the
+ * game. Should be less than MAX_POPULATION * number of players.
+ * @type {number}
+ */
+Genetics.Cage.START_MICE_PER_PLAYER = 2;
+
+/**
+ * Mapping of player id to and array with player name and code.
+ * @type {!Object.<number, Array.<string>>}
+ */
+Genetics.Cage.PLAYERS = {};
+
+/**
+ * Whether the simulation is running without a visualization.
+ * @type {boolean}
+ */
+Genetics.Cage.HEADLESS = false;
+
+/**
+ * List of mice currently alive.
+ * @type {!Array.<Genetics.Mouse>}
+ * @private
+ */
+Genetics.Cage.aliveMice_ = [];
```

Comment:
I store this also so I can pass it to a mouse interpreter easily instead of needing to creating a new list every time using miceMap and for easier iteration across all alive mice. Storing ids wouldn't be as useful. I'm still leaning towards using this, but I can instead remove this list and use miceMap everywhere.